### PR TITLE
Add a SWAP* test that checks in-place swaps

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR tests an uncommon edge case we do not currently cover with tests.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: pyvrp/search/tests/test_SwapStar.py [hunk 1]
```diff
@@ -71,3 +80,54 @@ def test_swap_star_on_RC208_instance(seed: int):
...
+    swap_star = SwapStar(data)
+
+    # Best is to exchange clients 1 and 3. The cost delta is all distance: it
+    # saves one expensive arc of cost 10, by replacing it with one of cost 1.
+    assert_equal(swap_star.evaluate(route1, route2, cost_eval), -9)
+
+    # Apply the move and test that it indeed swaps the nodes correctly.
+    swap_star.apply(route1, route2)
+    assert_(nodes[1].route is route2)
+    assert_(nodes[3].route is route1)
```
Comment (N-Wouda) [inline pyvrp/search/tests/test_SwapStar.py:133]: I'm happy that we can do these kinds of "surgical" tests on the operators now.
